src/rp2_common/pico_float/include/pico/float.h

```diff
@@ -21,66 +21,294 @@ extern "C" {
+* floating point implementation; they are however usually fine for the majority of cases
+* \endif
 *
-* On RP2350 RISC-V, only a small number of compiler runtime functions are overridden with faster implementations:
+* \if rp2350_specific
+* On Arm on RP2350, there are multiple options for `pico_float_pico`:
 *
-* - __addsf3, __subsf3, __mulsf3
+* 1. `pico_float_pico_vfp` - this library leaves basic C single-precision floating point operations to the compiler
+* which can use inlined VFP (Arm FPU) code. Custom optimized versions of trigonometric and scientific functions are provided.
+* no DCP (RP2350 Double co-processor) instructions are used.
+* 2. `pico_float_pico_dcp` - this library prevents the compiler injecting inlined VFP code, and also implements
+* all single-precision floating point operations in optimized DCP or M33 code. This option is not as fast
+* as pico_float_pico_vfp, however allows floating point operations without enabling the floating point co-processor
+* on the CPU; this can be beneficial in certain circumstances, e.g. where leaving stack in tasks or interrupts
+* for the floating point state is undesirable.
+*
+* Note: `pico_float_pico` is equivalent ot `pico_flot_pico_vfp` on RP2350, as this is the most sensible default
+* \endif
+*
+* On Arm, (replacement) optimized implementations are provided for the following compiler built-ins when
+* and math library functions when using `_pico` variants of `pico_float`:
+*
+* - basic arithmetic: (except `pico_float_pico_vfp`)
+*
+*   __aeabi_fadd, __aeabi_fdiv, __aeabi_fmul, __aeabi_frsub, __aeabi_fsub
+*
+* - comparison: (except `pico_float_pico_vfp`)
+*
+*   __aeabi_cfcmpeq, __aeabi_cfrcmple, __aeabi_cfcmple, __aeabi_fcmpeq, __aeabi_fcmplt, __aeabi_fcmple, __aeabi_fcmpge, __aeabi_fcmpgt, __aeabi_fcmpun
+*
+* - (u)int32 <-> float: (except `pico_float_pico_vfp`)
+*
+*    __aeabi_i2f, __aeabi_ui2f, __aeabi_f2iz, __aeabi_f2uiz
+*
+* - (u)int64 <-> float: (except `pico_float_pico_vfp`)
+*
+*   __aeabi_l2f, __aeabi_ul2f, __aeabi_f2lz, __aeabi_f2ulz
+*
+* - float -> double: (except `pico_float_pico_vfp`)
+*
+*   __aeabi_f2d
+*
+* - basic trigonometric:
+*
+*   sqrtf, cosf, sinf, tanf, atan2f, expf, logf
+*
+* - trigonometric and scientific
+*
+*   ldexpf, copysignf, truncf, floorf, ceilf, roundf, asinf, acosf, atanf, sinhf, coshf, tanhf, asinhf, acoshf, atanhf, exp2f, log2f, exp10f, log10f, powf, hypotf, cbrtf, fmodf, dremf, remainderf, remquof, expm1f, log1pf, fmaf
+*
+* - GNU exetnsions:
+*
+*   powintf, sincosf
+*
+* On Arm, the following additional optimized functions are also provided (when using `_pico` variants of `pico_float`):
+*
+* - Conversions to/from integer types:
+*
+*   - (u)int -> float (round to nearest):
+*
+*     int2float, uint2float, int642float, uint642float
+*
+*     note: on `pico_float_vfp` the 32-bit functions are also provided as C macros since they map to inline VFP code
```

Comment:
Are the mentions of `pico_float_vfp` here (and elsewhere) supposed to be `pico_float_pico_vfp` ? :shrug: